src/neural/onnx/network_onnx.cc

```diff
@@ -196,12 +196,11 @@ Ort::Value OnnxComputation<DataType>::PrepareInputs(int start, int batch_size) {
   int end = std::min(start + batch_size, static_cast<int>(raw_input_.size()));
   for (int i = start; i < end; i++) {
     for (const auto& plane : raw_input_[i]) {
+      DataType value = std::is_same<Ort::Float16_t, DataType>::value
+                           ? FP32toFP16(plane.value)
+                           : plane.value;
       for (auto bit : IterateBits(plane.mask)) {
-        if (std::is_same<Ort::Float16_t, DataType>::value) {
```

Comment:
btw if the previous version, `if constexpr` would ensure it's evaluated at the compile time (it worked without it too, though).

Reply:
I did this for the `FP32toFP16()` call, I'm not sure MSVC can figure out it can be pulled out of the loop.